src/main/java/org/mskcc/limsrest/service/ilabs/GetGeneralInfo.java

```diff
@@ -0,0 +1,365 @@
+        String piLastName = labPi.get("last_name").asText();
+        String piEmail = labPi.get("email").asText();
+
+        // check if we should override this data
+        if (CORRECTED_PIS.containsKey(labId)) {
+            String s = CORRECTED_PIS.get(labId);
+            String[] temp = s.split(",");
+            piEmail = temp[0];
+            piLastName = temp[1];
+            if (temp.length > 2) {
+                piFirstName = temp[2];
+            }
+        }
+
+        researcher.setFirstName(piFirstName);
+        researcher.setLastName(piLastName);
+        researcher.setEmail(piEmail);
+        return Optional.of(researcher);
+    }
+
+    private List<CustomForm> parseCustomForms(String url, RestTemplate restTemplate) {
+        ObjectNode customFormsJson = restTemplate.getForObject(url, ObjectNode.class);
+        List<CustomForm> parsedCustomForms = new ArrayList<>();
+        JsonNode arrayNode = customFormsJson.get("ilab_response").get("custom_forms");
+        if (arrayNode == null || arrayNode.size() == 0) {
+            throw new RuntimeException("Could not get custom form, check to see if API changed");
+        }
+
+        Iterator<JsonNode> iterator = arrayNode.iterator();
+        while (iterator.hasNext()) {
+            JsonNode customForm = iterator.next();
+            if (ObjectUtils.isEmpty(customForm.get("fields"))) continue;
+            String id = customForm.get("id").asText();
+            String name = customForm.get("name").asText();
+            String note = customForm.get("note").asText();
+            CustomForm parsedCustomForm = new CustomForm(id, name, note);
+            JsonNode fields = customForm.get("fields");
+            fields.forEach(field -> {
+                if (field.get("value") != null) {
+                    String lcFormName = field.get("name").asText();
+//                    System.out.println("parsecustom " + lcFormName);
+                    String lcFormValue = "";
+                    if (field.get("value").isArray()) {
+                        List<String> vals = new ArrayList<>();
+                        field.get("value").forEach(jsonNode -> vals.add(jsonNode.asText()));
+                        lcFormValue = String.join(";", vals);
+                    } else {
+                        lcFormValue = field.get("value").asText();
+                    }
+//                    System.out.println(lcFormValue);
+                    parsedCustomForm.addField(lcFormName, lcFormValue);
+                }
+            });
+            parsedCustomForms.add(parsedCustomForm);
+        }
+        return parsedCustomForms;
+    }
+
+
+    public void printMapping(Map<String, String> ilabValues) {
+        for (Map.Entry<String, String> entry : ilabValues.entrySet()) {
+            System.out.println(entry.getKey() + ":");
+            System.out.println("  " + entry.getValue());
+        }
+        System.out.println();
+
+    }
+
+    @Deprecated
```

Comment:
Do we need this deprecated function?

Reply:
This function has old unused Ruby code so I removed it.